src/Compilers/CSharp/Portable/Symbols/OverriddenOrHiddenMembersHelpers.cs

```diff
@@ -938,9 +939,15 @@ internal static bool RequiresExplicitOverride(this MethodSymbol method, out bool
             if (!methodimplWouldBeAmbiguous)
                 return true;
 
+            Debug.Assert(runtimeOverriddenMethod is { });
+
             // We produce the warning when a methodimpl would be required but would be ambiguous to the runtime.
-            // This ambiguity only arises due to collision after generic substitution.
-            warnAmbiguous = csharpOverriddenMethod.ContainingType.IsGenericType;
+            // However, if there was a duplicate definition for the runtime signature of the overridden
+            // method where it was originally delcared, that would have been an error.  In that case we suppress
+            // the warning as a cascaded diagnostic.
+            bool originalOverriddenMethodWasAmbiguious =
+                csharpOverriddenMethod.IsDefinition || csharpOverriddenMethod.OriginalDefinition.MethodHasRuntimeCollisionInSubstututedForm();
```

Comment:
>csharpOverriddenMethod.IsDefinition || csharpOverriddenMethod.OriginalDefinition.MethodHasRuntimeCollisionInSubstututedForm(); [](start = 16, length = 126)

I think that placing logic here in this form can actually dismiss true ambiguities that can exist in substituted form at the same time. I think solution suggested in the previous PR would handle that better. Please add tests for scenarios like that.

Reply:
This test is intended to dismiss warnings in cases in which there are necessarily true errors that have already been reported. In other words, the intent is to suppress diagnostics that are cascaded. The suggestion in the previous PR fails to eliminate a number of scenarios of cascaded diagnostics that are eliminated by this approach.

---
In reply to: [452982335](https://github.com/dotnet/roslyn/pull/45849#discussion_r452982335) [](ancestors = 452982335)

Reply:
> The suggestion in the previous PR fails to eliminate a number of scenarios of cascaded diagnostics that are eliminated by this approach.

Could you please share the details about the scenarios? I have high degree of confidence that the suggestion can be easily accommodated to handle them too.

---
In reply to: [453768445](https://github.com/dotnet/roslyn/pull/45849#discussion_r453768445) [](ancestors = 453768445,452982335)

Reply:
We are not reviewing a different PR. We are reviewing this one. The tests supporting this PR are included in this PR. Can you please review them?

---
In reply to: [453798022](https://github.com/dotnet/roslyn/pull/45849#discussion_r453798022) [](ancestors = 453798022,453768445,452982335)